es-vue-base/src/lib-components/EsFileUpload.vue

```diff
@@ -0,0 +1,230 @@
+<!-- eslint-disable vuejs-accessibility/no-static-element-interactions -->
+<template>
+    <div
+        class="es-file-upload align-items-center rounded d-flex flex-column justify-content-center p-2"
+        :class="{ 'active': active }"
+        @dragenter.stop.prevent="active = true"
+        @dragover.stop.prevent="active = true"
+        @dragleave.stop.prevent="active = false"
+        @dragend.stop.prevent="onDrop"
+        @drop.stop.prevent="onDrop">
+        <div
+            class="d-flex justify-content-center"
+            :class="{ 'flex-row': collapsed, 'flex-column': !collapsed }">
+            <div class="align-self-center p-2">
+                <icon-upload
+                    class="d-none d-md-inline-block mb-4"
+                    height="48px"
+                    width="48px" />
+            </div>
+            <div class="align-self-center p-2">
+                <slot name="cta" />
+            </div>
+            <div class="align-self-center p-2">
+                <es-button
+                    class="d-none d-md-inline-block w-35 mb-4"
+                    variant="secondary"
+                    outline
+                    @click="openFilePicker">
+                    Browse Files
+                </es-button>
+                <es-button
+                    class="d-inline-block d-md-none w-100 mb-2"
+                    variant="secondary"
+                    outline
+                    @click="openFilePicker">
+                    Choose File
+                </es-button>
+            </div>
+        </div>
+        <div>
+            <slot name="helpText" />
+        </div>
+        <b-form-file
+            ref="fileInput"
+            v-model="pickedItems"
+            :state="Boolean(pickedItems)"
+            :accept="fileTypesAsString"
+            style="display: none"
+            multiple />
+    </div>
+</template>
+
+<script lang="js">
+import { BFormFile } from 'bootstrap-vue';
+import EsButton from './EsButton.vue';
+import IconUpload from '../lib-icons/icon-upload.vue';
+
+export default {
+    name: 'EsFileUpload',
+    components: {
+        BFormFile,
+        EsButton,
+        IconUpload,
+    },
+    props: {
+        uploadUrls: {
+            type: Array,
+            default: () => [],
+        },
+        fileTypes: {
+            type: Array,
+            default: () => [],
+        },
+        maxFileSize: {
+            type: Number,
+            default: 100,
```

Comment:
`100` [what](https://en.wikipedia.org/wiki/Magic_number_(programming))? KB? MB?

Reply:
MB! Added a reasonable default and renamed to `maxFileSizeMb`